app/src/main/java/com/example/fivenightsescape/player/Player.kt

```diff
@@ -1,18 +1,22 @@
 package com.example.fivenightsescape.player
 
 import android.location.Location
+import android.widget.TextView
 import androidx.lifecycle.ViewModel
 import com.google.android.gms.maps.model.LatLng
 
-const val PLAYER_HEALTH = 3
+private const val PLAYER_HEALTH = 3
 
-class Player(var position: LatLng): ViewModel() {
+class Player(var position: LatLng, val healthBar: TextView): ViewModel() {
```

Comment:
This field leaks a context object.
You can fix it by changing it to:
```kt
class Player(var position: LatLng, healthBar: TextView) : ViewModel() {
  private var health: Int = PLAYER_HEALTH // This line stays the same
  private val healthBarRef: WeakReference<TextView> = WeakReference(healthBar)
  private val healthBar: TextView?
  get() = healthBarRef.get()
```
add `?` to both `this.healthBar.text = this.health.toString()`:
```kt
this.healthBar?.text = this.health.toString()
```